include/oneapi/dpl/internal/reduce_by_segment_impl.h

```diff
@@ -22,6 +22,7 @@
 #include "../pstl/iterator_impl.h"
 #include "function.h"
 #include "by_segment_extension_defs.h"
+#include "../ranges"
```

Comment:
As I understand we would have both iterator-based and range-based version of the API. What I don't like is our iterator based version that is **not** in `experimental` namespace relies on something that **is** `experimental` namespace. I believe we already have such precedence but we should be really careful with that.

And furthermore, our iterator-based version relies on hardcoded constant in the range-based implementation that concerns me much more than relying on experimental API.

Reply:
I've added the workgroup computation that replaces the hardcoded constant. In this case use of `experimental` features is needed to provide the performance benefit to all interfaces.  I agree with you caution is to be used, and in this case I think the use is justified by the benefit.

Reply:
BTW, what about performance with  the "workgroup computation"?

Reply:
1. I'll run those tests today.
2. I agree we need to do that experiment. I will add the TODO.

Reply:
@MikeDvorskiy, performance with the "workgroup computation" is slightly better than with the hardcoded constants. The algorithm is a little more than 2x faster than the algorithm we were using in the iterator-based API.